code/game/machinery/doors/_door.dm

```diff
@@ -396,49 +396,57 @@
 /obj/machinery/door/proc/open(var/forced = 0)
 	if(!can_open(forced))
 		return
-	operating = 1
+	operating = TRUE
 
 	do_animate("opening")
 	icon_state = "door0"
-	set_opacity(0)
-	sleep(3)
-	src.set_density(0)
+	set_opacity(FALSE)
+	sleep(0.3 SECONDS)
+	src.set_density(FALSE)
 	update_nearby_tiles()
-	sleep(7)
+	addtimer(CALLBACK(src, .proc/open_final), 0.7 SECONDS, TIMER_UNIQUE)
```

Comment:
Am a bit confused; why do we _both_ sleep and then create a timer? Won't one of them have the same issue as you're trying to resolve in the current code (and which in theory should be resolved already via the `operating` var?

Reply:
I am not really sure we need timer here. I would remove it completely for doors, because we have this 2 sleep logic 
(If door is slightly opened, we can already move through it)